pkg/controllers/dynakube/logmonitoring/logmonsettings/reconciler.go

```diff
@@ -37,7 +37,7 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 		return nil
 	}
 
-	if !r.dk.LogMonitoring().IsEnabled() || r.dk.Status.KubernetesClusterMEID == "" {
+	if !r.dk.LogMonitoring().IsEnabled() {
 		meta.RemoveStatusCondition(r.dk.Conditions(), conditionType)
 
 		if r.dk.Status.KubernetesClusterMEID == "" {
```

Comment:
you should move the check:
```
		if r.dk.Status.KubernetesClusterMEID == "" {
			return errors.New("the status of the DynaKube is missing information about the kubernetes monitored-entity, skipping LogMonitoring deployment")
		}

```
after the `!r.dk.LogMonitoring().IsEnabled()`

like:
```
if !r.dk.LogMonitoring().IsEnabled() {
	... // has early return
} else if r.dk.Status.KubernetesClusterMEID == "" { 
	return errors.New("the status of the DynaKube is missing information about the kubernetes monitored-entity, skipping LogMonitoring deployment"
}
```

Reply:
You're right as this should be a separate case. Thanks, will do :)

Reply:
done https://github.com/Dynatrace/dynatrace-operator/pull/4205/commits/93f164b0ff1bf116da05052bcc7aebd52c7bb851

@andriisoldatenko please check if the test runs through. Locally for me it ran through successfully several times

Reply:
it works!